File: services/api/app/api/schemas/alerts.py
from typing import List, Literal, Optional
from pydantic import BaseModel

AlertSeverity = Literal["high", "medium", "low"]
AlertType = Literal[
    "Heavy Rainfall",
    "Regime Transition",
    "High Uncertainty",
    "Flash Flood Risk",
    "Rapid Moisture Advection"
]

class OperationalAlert(BaseModel):
    id: str
    type: str
    gridId: str
    location: str
    detail: str
    timeAgo: str
    severity: AlertSeverity
    lat: Optional[float] = None
    lon: Optional[float] = None

class AlertFeedResponse(BaseModel):
    timestamp: str
    activeCount: int
    criticalCount: int
    alerts: List[OperationalAlert]

File: frontend/types/alerts.ts
import { AlertType } from "@/components/cards/alert-card";

export type AlertSeverity = "high" | "medium" | "info";

export interface OperationalAlert {
  id: string;
  type: AlertType;
  gridId: string;
  location: string;
  detail: string;
  timeAgo: string;
  severity: AlertSeverity;
  lat?: number;
  lon?: number;
}

export interface AlertFeedResponse {
  timestamp: string;
  activeCount: number;
  criticalCount: number;
  alerts: OperationalAlert[];
}
